src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ManagedGroupRoutes.scala

```diff
@@ -116,7 +116,8 @@ trait ManagedGroupRoutes extends SamUserDirectives with SecurityDirectives with
       samRequestContext: SamRequestContext): Route =
     requireAction(managedGroup, SamResourceActions.sharePolicy(accessPolicyName), samUser.id, samRequestContext) {
       withSubject(WorkbenchEmail(email), samRequestContext) { subject =>
-        complete(managedGroupService.addSubjectToPolicy(managedGroup.resourceId, accessPolicyName, subject, samRequestContext).map(_ => StatusCodes.NoContent))
+        complete(managedGroupService.addSubjectToPolicy(managedGroup.resourceId, accessPolicyName, subject, samRequestContext)
+          .map(if (_) StatusCodes.NoContent else StatusCodes.NotFound))
```

Comment:
I don't think this does what you want. The authorization check is the `requireAction` call above. `addSubjectToPolicy` when no change was actually made (i.e. member was already in the groups), not when the caller lacks access.